elements-fun/src/transaction.rs

```diff
@@ -384,20 +544,167 @@ pub struct PegoutData<'txo> {
     pub extra_data: Vec<&'txo [u8]>,
 }
 
+// TODO: think about tagging
 /// Transaction output
-#[derive(Clone, Default, PartialEq, Eq, Debug, Hash)]
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(crate = "serde_crate")
+)]
+pub enum TxOut {
+    Null(NullTxOut),
+    Explicit(ExplicitTxOut),
+    Confidential(ConfidentialTxOut),
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(crate = "serde_crate")
+)]
+pub struct NullTxOut {
```

Comment:
:question: When does elements use null txouts? Just curious.

Reply:
I don't fully understand this yet :D